code/Gamemodes/Bowling/Player/BowlingPlayer.cs

```diff
@@ -19,30 +28,50 @@ public override void Respawn()
 
 		// This should be fired from the state handler once implemented.
 		OnTurnStarted();
+
+		Tags.Add( "bowling_owner" );
 	}
 
 	/// <summary>
 	/// Called when this players turn has started.
 	/// </summary>
 	public void OnTurnStarted()
 	{
+		Log.Debug( $"Bowling turn started for: {Client.Name}" );
+
 		// TODO: Face your assigned alley once your turn begins. This is needs to be replaced later when we have alley entities.
 		Rotation = Rotation.FromYaw( 90 );
 
-		ActiveChild = new BowlingBallCarriable();
-		ActiveChild.OnCarryStart( this );
+		if ( ActiveChild is not BowlingBallCarriable )
+		{
+			ActiveChild = new BowlingBallCarriable();
+			ActiveChild.OnCarryStart( this );
+		}
+
+		// reset draw
+		(ActiveChild as BowlingBallCarriable).Reset();
```

Comment:
Could be null (?)

Reply:
it gets initialized right above on a check, right? I suppose it could technically be not valid though so I'll throw that in there as well